flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/SlotID.java

```diff
@@ -75,9 +75,15 @@ public int hashCode() {
 
 	@Override
 	public String toString() {
-		return "SlotID{" +
-			"resourceId=" + resourceId +
-			", slotId=" + slotId +
-			'}';
+		return resourceId + "_" + slotId;
+	}
+
+	/**
+	 * Generate a random slot id.
+	 *
+	 * @return A random slot id.
+	 */
+	public static SlotID generate() {
+		return new SlotID(ResourceID.generate(), 0);
```

Comment:
A SlotID is a identify of the slot which locates in some TaskManager. In current flink's implementaion, we use integer to represent a slot within TM, i think we should make this more explicit.
For generating a random slot, i think random a resource id which slot is located is enough. (We may have some limitation of the integer else where, like the integer can not be greater than the total slot number of the TM, so i choose 0)